src/main/java/com/eerussianguy/blazemap/feature/maps/AtlasExporter.java

```diff
@@ -44,26 +45,31 @@ private static synchronized void resetTask() {
     }
 
     private static void exportAsync() {
-        task.setStage(Task.Stage.CALCULATING);
-        TileResolution resolution = TileResolution.FULL;
-        StorageAccess.Internal storage = BlazeMapClientEngine.getDimensionStorage(task.dimension);
-        NativeImage atlas = constructAtlas(storage, resolution);
+        NativeImage atlas = null;
 
-        try(atlas) {
+        try {
+            // Calculation stage: figure out image size and corner regions, create canvas of appropriate size
+            task.setStage(Task.Stage.CALCULATING);
+            TileResolution resolution = TileResolution.FULL;
+            StorageAccess.Internal storage = BlazeMapClientEngine.getDimensionStorage(task.dimension);
+            atlas = constructAtlas(storage, resolution);
+
+            // Stitching stage: open tiles and transfer pixels to the correct place in the atlas
             task.setStage(Task.Stage.STITCHING);
             var layerKeys = task.map.value().getLayers();
-            for(var layerKey : layerKeys) {
+            for(var layerKey : layerKeys) { // Loop layers
                 if(!task.layers.contains(layerKey)) continue;
                 File folder = storage.getMipmap(layerKey.location, ".", resolution);
-                for(int regionX = task.atlasStartX; regionX <= task.atlasEndX; regionX++) {
+                for(int regionX = task.atlasStartX; regionX <= task.atlasEndX; regionX++) { // Loop regions
```

Comment:
Could we have per-region error handling, falling back to just skipping that region if it can't access that tile? Better to have 95% of an export than no export!

Reply:
Done